### ISSUE TITLE
In Snow Trouble!, enemies fall infinitely looks like rain.

### ISSUE THREAD

**Issue by Jmousy** [ruffle-rs/ruffle#4122]
**Describe the bug**
![snow](https://user-images.githubusercontent.com/75052171/115993315-3510a100-a60d-11eb-81f1-331361f6bd32.gif)
(It is explained in the gif attached at the top.)

When the game starts in Snow Trouble!, enemies do not land correctly on the platform and keep dropping. Because of this, the game cannot be played.

If you control the player with the arrow keys, the player will fall off the stage and become invisible forever.

[Snow Trouble!.zip](https://github.com/ruffle-rs/ruffle/files/6371920/Snow.Trouble.zip)

**Is the problem with the Ruffle desktop app, extension, or self-hosted version?**
Checked in self-hosted version.

**What platform are you using?**
 - Windows 10 20H1 x64
 - Google Chrome (latest)
 - Ruffle Nightly 2021-04-25

**Comment by Toad06** [comment]
Thank you for the report.

---

Collisions seem to be done in a unusual way. Most collision detections use the following function:

```
function checkColour(c, x, y)
{
   /* this part is in the global scope in the file
   var snappy = new flash.display.BitmapData(2000,600);
   snappy.draw(_root.screen.maskset,new flash.geom.Matrix());
   _root.snapshot = snappy;
   */
   colour = snapshot.getPixel(x,y).toString(16);
   trace(colour); // added for debugging
   if(!char.falling)
   {
   }
   if(colour == c)
   {
      return true;
   }
   return false;
}
```

When running the game without moving the character, Flash Player returns the following strings: "ff", "ffff00" and "ffffff".
But in the same situation, Ruffle returns "ffffff" and "0".

Because these results don't match, there are issues when trying to detect collisions in Ruffle:
`if(checkColour("ff",ob.clip._x,ob.clip._y,true))`
`if(!checkColour("ffff00",ob.clip._x,ob.clip._y - _loc2_,false))`

Note that this is not an issue with the `.toString(16)` call, as `snapshot.getPixel(x,y);` gives the same results:
- Flash: 255 (0xff), 16776960 (0xffff00), 16777215 (0xffffff).
- Ruffle: 16777215 (0xffffff), 0 (0x0)

---

So I think these collision issues should go away once `BitmapData.draw` gets implemented.

**Comment by relrelb** [comment]
Related to #290.

**Comment by relrelb** [comment]
Fixed by #7254.

**Comment by Lord-McSweeney** [comment]
The issue is fixed by #9134, see #9071 for another issue with the game
